packages/app-store/_utils/writeAppDataToEventType.ts

```diff
@@ -0,0 +1,57 @@
+import { EventTypeRepository } from "@calcom/lib/server/repository/eventType";
+import prisma from "@calcom/prisma";
+import type { AppCategories } from "@calcom/prisma/enums";
+import { EventTypeMetaDataSchema } from "@calcom/prisma/zod-utils";
+
+import type { appDataSchemas } from "../apps.schemas.generated";
+
+const writeAppDataToEventType = async ({
```

Comment:
Just realized that this could cause a lot of eventType updates in parallel possibly impacting DB.

On a CRM app's install, we are updating all existing event-types to set `enabled:false` to get the new behaviour of no-automatic CRM connection from event-type. But if we create a new event-type, after installing the new CRM app, it would still have the CRM app automatically enabled for that event. This is because appMetadata isn't set for that event-type and we end up thinking that the app should be considered enabled.	


What if we identify that the installation is a new one and then decide the behaviour from that. **In this way, we don't need to update all event-types and the bug I mentioned above is also fixed automatically.**

One approach could be to check if `credential.type.endsWith('_crm').` I know that the change that switched to `_crm` has been live for some time but it could be very few people having installed the crm apps after that. We could run a migration for such users maybe to move them back to `_other_calendar`.

But the point is that once I know that the credential is a new installation, we can then simply decide based on that if we need to consider the default value of `enabled`(when it is undefined) to be true or false.


This should remove the requirement of calling this function.